services/venice-router/src/main/java/com/linkedin/venice/router/api/VeniceVersionFinder.java

```diff
@@ -88,9 +88,13 @@ public int getVersion(String storeName, BasicFullHttpRequest request) throws Ven
 
     int metadataCurrentVersion = store.getCurrentVersion();
     if (!lastCurrentVersionMap.containsKey(storeName)) {
```

Comment:
Not in this change, but it looks like there could have a race condition from L90 - L102, i.e., if multiple threads discover that `lastCurrentVersionMap` doesn't contain the key at same time and, all of them go ahead to refresh and update the K-V at the same time. This is because the test and set operations here is not atomic, however, I don't know if this could be a benign one or not, but just a heads up.

In case if we want to fix the race and get a peaceful mind, we can probably wrap them into a `computeIfAbsent` block.